include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -0,0 +1,352 @@
+ *
+ */
+class sequence_file_format_embl
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+     //!\brief Default constructor is explicitly deleted, you need to provide a stream or file name.
+    sequence_file_format_embl() = default;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the embl file.
+    sequence_file_format_embl(sequence_file_format_embl const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the embl file.
+    sequence_file_format_embl & operator=(sequence_file_format_embl const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_format_embl(sequence_file_format_embl &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_format_embl & operator=(sequence_file_format_embl &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "embl" },
+    };
+
+    //!\copydoc sequence_file_input_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                            & stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                                                               & sequence,
+              id_type                                                                & id,
+              qual_type                                                              & SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+        auto stream_it = ranges::begin(stream_view);
+
+        std::string identifier;
+        ranges::copy(stream_view | view::take_until_or_throw(is_cntrl || is_blank),
+                     std::back_inserter(identifier));
+        if (identifier != "ID")
+            throw std::logic_error{"An entry has to start with the code word ID."};
+
+        if (options.complete_header)
+        {
+            while (identifier != "SQ")
+            {
+                if constexpr (!detail::decays_to_ignore_v<id_type>)
+                {
+                    ranges::copy(identifier  | view::char_to<value_type_t<id_type>>,
+                                 std::back_inserter(id));
```

Comment:
Why do you copy the "ID" identifier into the id string for every line? I suggest only to copy the text itself; without any format-specific stuff. Imagine, you use the `id` and `sequence` variables for printing FASTA...